go-auth/google/spec.go

```diff
@@ -0,0 +1,24 @@
+package google_auth
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	pf "github.com/estuary/flow/go/protocols/flow"
+)
+
+func Spec(scopes ...string) *pf.OAuth2Spec {
+	authUrlTemplate := fmt.Sprintf(
+		"https://accounts.google.com/o/oauth2/auth?access_type=offline&prompt=consent&client_id={{ client_id }}&redirect_uri={{ redirect_uri }}&response_type=code&scope=%s&state={{ state }}",
```

Comment:
Somewhat surprisingly, the auth URL does not work with those swapped. It might have something to do with the way our stack uses the format strings, since I can't think of a reason why the order of query parameters would matter. I brought the format string out as a constant though, but left those values where they were.